app/main.py

```diff
@@ -1,13 +1,21 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {"line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"],
+            "name": error["code"],
+            "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [format_linter_error(error)
+            for error in errors if error["filename"] == file_path],
+            "path": file_path,
+            "status": "failed"}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [{"errors":
+             format_single_linter_file(source, errors).get("errors", []),
+             "path": source,
+             "status": "failed" if len(errors) else "passed"}
+            for source, errors in linter_report.items()]
```

Comment:
You can simplify this method
```suggestion
    return [ format_single_linter_file(file_path, errors)
        for file_path, errors in linter_report.items()]
```